### PR TITLE
[Console] Fix clear line with question in section

### LINKED ISSUE #47411: [Console] Question helper is behaving differently when inside a Section
### Symfony version(s) affected

6.1.4

### Description

The question helper is going onto a new line when used inside a Section, therefore section cannot rewrite correctly

### How to reproduce

# Question helper only
<details>
  <summary>Code</summary>
  
  ```php
        $io = new SymfonyStyle($input, $output);

        $countdown = 3;
        while ($countdown > 0) {
            $io->write('Lorem ipsum ' . $countdown);
            $random = $io->ask('Press random key');
            $countdown--;
        }
  ```
</details>
<a href="https://asciinema.org/a/517552" target="_blank"></a>

# Section helper only
<details>
  <summary>Code</summary>
  
  ```php
        $section = $output->section();

        $countdown = 3;
        while ($countdown > 0) {
            $section->clear();
            $section->write('Lorem ipsum ' . $countdown);
            $countdown--;
            sleep(1);
        }
  ```
</details>
<a href="https://asciinema.org/a/517554" target="_blank"></a>

# Both combined
<details>
  <summary>Code</summary>
  
  ```php
        $section = $output->section();
        $io = new SymfonyStyle($input, $section);

        $countdown = 3;
        while ($countdown > 0) {
            $section->clear();
            $section->write('Lorem ipsum ' . $countdown);

            $random = $io->ask('Press random key');
            $countdown--;
        }
  ```
</details>
<a href="https://asciinema.org/a/517555" target="_blank"></a>

### Expected
Final output should be
```
Lorem ipsum 1

 Press random key:
 > something

```

### PR DESCRIPTION
**Pull request by maxbeckers**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #47411
| License       | MIT
| Doc PR        | -

In the issue #47411 is the current behavior described (with videos).

The problem is in a section using a question and a clear. Then one line is not cleared because of the `return` so submit the input.

NOTICE: This bug might be as well in the versions 4.4+, but in the versions < 6.1 it would be more complicated to fix, because the `SymfonyStyle` does not have the property `$output` in this versions.

### CODE REVIEW

**Comment by bhujagendra-ishaya** [comment]
Hi @maxbeckers 

Thanks for the provided fix.

To be able to easier review the fix, would you or @bigfoot90 mind adding a full working code (complete file with imports and stuff) to here or the original issue https://github.com/symfony/symfony/issues/47411? - I'd rather also run the code, than just the tests.

Also, did you consider to include your change mentioned in comment https://github.com/symfony/symfony/issues/47411#issuecomment-1298594094?

**Comment by maxbeckers** [comment]
@bhujagendra-ishaya thanks for your comment.

The change mentioned in the comment can be igored, because it's not a problem. It was my fault because of any issues in my vendor.

here is the reproducer with the fix in vendors:
[test-48089.zip](https://github.com/symfony/symfony/files/9937499/test-48089.zip)

Just run on CLI: `php testcommand.php` in the root.

**Review by bhujagendra-ishaya — APPROVED** [review]
Ok, I've tested your provided scirpt ...

1. with your provided vendor dir
2. with a freshly installed vendor dir

... on the following two environments:

```
Microsoft Windows [Version 10.0.19044.2130]

> php -v
PHP 8.1.12 (cli) (built: Oct 25 2022 18:20:48) (NTS Visual C++ 2019 x64)
Copyright (c) The PHP Group
Zend Engine v4.1.12, Copyright (c) Zend Technologies
```

... and ...

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy

$ php -v
PHP 8.1.12 (cli) (built: Oct 28 2022 17:39:57) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.12, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.12, Copyright (c), by Zend Technologies
```

I can confirm that *in both environments*, ...

- the issue **happens** with `symfony/console v6.1.7`,
- the issue **is fixed** with `symfony/console v6.1.7 + fix applied`.

**Review by nicolas-grekas — APPROVED** [review]
Not sure I precisely understand what's going on, but LGTM if you tested and it fixes the issue.

**Comment by maxbeckers** [comment]
@nicolas-grekas did you see the videos in the issue describing the problem in detail.

**Comment by maxbeckers** [comment]
@nicolas-grekas the `ConsoleSectionOutput` is counting it's lines. But when there is a input in the section and the input is confirmed with ENTER, the new line of the ENTER was not added to the section. This is what this fix is doing. Hope this helps to understand this fix.

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thanks for improving sections support, @maxbeckers. That's much appreciated.

**Comment by chalasr** [comment]
PR welcome on 5.4 if you find a way to have the same behavior there.

**Comment by maxbeckers** [comment]
@chalasr I could try a fix after my vacation time in the beginning of the next year, but i don't know if it's worth the effort. Because the fix will then be a way more complex.

**Comment by chalasr** [comment]
@maxbeckers I guess we can live with that. Thanks!

**Comment by nicolas-grekas** [comment]
@maxbeckers would you still have time to check the failures on branch 6.2 that came up after merging this PR up?

**Comment by maxbeckers** [comment]
@nicolas-grekas sorry, but since today im in vacation and have only my smartphone. Feel free to revert the change and i'll check it after holiday.

UPDATE: This is the new test failing ... but the test output is looking to me a bit strange. Why is there `foo` twice? Maybe a bug in 6.2? I'd like to investigate some time but i can only do that afer my vacation in the beginning of next year. Maybe someone has the time to have a deeper look into and fixes it or you revert and i'll do it when im back.

**Comment by chalasr** [comment]
Thanks for the hint. I'll have a look so we can decide depending on the result. Enjoy your vacation :)

**Pull request merged**
